### ISSUE TITLE
[NativeAOT] Implement a compiler "optimize for application" switch analogous to /GA for MSVC

### ISSUE THREAD

**Issue by VSadov** [dotnet/runtime#87642]
The flag indicates that the compiled module is a singlefile executable and can drive some optimizations.

Single file executable is a common enough scenario that c++ compiler has a switch for it. ILC could benefit from such switch as well.

**Comment by NCLnclNCL** [comment]
> The flag indicates that the compiled module is a singlefile executable and can drive some optimizations.
> 
> Single file executable is a common enough scenario that c++ compiler has a switch for it. ILC could benefit from such switch as well.
> 
> Re: [#87148 (comment)](https://github.com/dotnet/runtime/pull/87148#discussion_r1226018085)

it will do reduce file size ??

**Comment by VSadov** [comment]
> it will do reduce file size ??

Yes. linker can replace more complex code for the dynamic case with more efficient instructions if it is known at link time that we are linking an executable. However, the size redundancy will still be there.

The size difference would vary between platforms. On some platforms like linux-x64 the size difference is relatively small, but on others it would be more noticeable: See for example https://github.com/dotnet/runtime/pull/87148/commits/f806f34cc4c4aab5865290b436bf88bd5fb9ffde#r1234559333
